cvl/internal/yparser/yparser.go

```diff
@@ -488,56 +602,40 @@ func (yp *YParser) MergeSubtree(root, node *YParserNode) (*YParserNode, YParserE
 	return (*YParserNode)(rootTmp), YParserError{ErrCode: YP_SUCCESS}
 }
 
-func (yp *YParser) DestroyCache() YParserError {
-
-	if yp.root != nil {
-		C.lyd_free_withsiblings((*C.struct_lyd_node)(yp.root))
-		yp.root = nil
-	}
-
-	return YParserError{ErrCode: YP_SUCCESS}
-}
-
-// SetOperation Set operation
-func (yp *YParser) SetOperation(op string) YParserError {
-	if ypOpNode == nil {
-		return YParserError{ErrCode: YP_INTERNAL_UNKNOWN}
-	}
+// createTempDepData merge depdata and data to create temp data. used in syntax, semantic and custom validation
+func (yp *YParser) mergeDepData(data *(*C.struct_lyd_node), depData *YParserNode, destruct bool) YParserError {
+	var flags C.uint16_t
 
-	if C.lyd_change_leaf_data((*C.struct_lyd_node)(ypOpNode), C.CString(op)) != 0 {
-		return YParserError{ErrCode: YP_INTERNAL_UNKNOWN}
+	flags = 0
+	if destruct {
+		flags |= C.LYD_MERGE_DESTRUCT
 	}
 
-	yp.operation = op
-	return YParserError{ErrCode: YP_SUCCESS}
-}
-
-// createTempDepData merge depdata and data to create temp data. used in syntax, semantic and custom validation
-func (yp *YParser) createTempDepData(dataTmp *(*C.struct_lyd_node), depData *YParserNode) YParserError {
-
-	if C.lyd_merge_to_ctx(dataTmp, (*C.struct_lyd_node)(depData), C.LYD_OPT_DESTRUCT, (*C.struct_ly_ctx)(ypCtx)) != 0 {
+	if C.lyd_merge_siblings(data, (*C.struct_lyd_node)(depData), flags) != C.LY_SUCCESS {
 		TRACE_LOG((TRACE_SYNTAX | TRACE_LIBYANG), "Unable to merge dependent data\n")
 		return getErrorDetails()
 	}
 	return YParserError{ErrCode: YP_SUCCESS}
 }
 
 // ValidateSyntax Perform syntax checks
-func (yp *YParser) ValidateSyntax(data, depData *YParserNode) YParserError {
-	dataTmp := (*C.struct_lyd_node)(data)
-
-	if data != nil && depData != nil {
-		//merge ependent data for synatx validation - Update/Delete case
-		err := yp.createTempDepData(&dataTmp, depData)
+func (yp *YParser) ValidateSyntax(data *YParserNode, depData *YParserNode) YParserError {
+	dataPtr := (*C.struct_lyd_node)(data)
+
+	if depData != nil {
+		// merge dependent data for syntax validation - Update/Delete case
+		// This is a destructive merge, depData is no longer valid.
+		err := yp.mergeDepData(&dataPtr, depData, true)
+		depData = nil
 		if err.ErrCode != YP_SUCCESS {
 			return err
 		}
 	}
 
 	//Just validate syntax
-	if C.lyd_data_validate(&dataTmp, C.LYD_OPT_EDIT|C.LYD_OPT_NOEXTDEPS, (*C.struct_ly_ctx)(ypCtx)) != 0 {
+	if C.lyd_validate_all(&dataPtr, (*C.struct_ly_ctx)(ypCtx), C.LYD_VALIDATE_PRESENT|C.LYD_VALIDATE_NO_STATE|C.LYD_VALIDATE_NOEXTDEPS, nil) != C.LY_SUCCESS {
 		if IsTraceAllowed(TRACE_ONERROR) {
```

Comment:
Can you point me to the patches to be applied on top of libyang3 master/devel branch to get LYD_VALIDATE_NOEXTDEPS? I will also try these changes on my testbed

Reply:
yes, that's it.  I also put the link in the description for the ticket.

Reply:
also, make sure you use the 'devel' branch.  Master needs a few more patches: https://github.com/bradh352/sonic-buildimage/tree/bradh352/libyang3-only/src/libyang3/patch